Fixes Ram bugs reported in #1363 and #1377 and a bug in writing the open and constant maps to the .circ file

BFH-ktt1 opened the pull request.

MarcinOrlowski commented on src/main/java/com/cburch/logisim/fpga/data/MapComponent.java:
```diff
@@ -651,6 +652,7 @@ public String getDisplayString(int pin) {
           ioConst |= constants.get(i) & 1;
         }
       }
+      if (pin < -1) return s.toString();
```
can we have magic value "-2" replaced with constant? If not or it makes no sense then please add the comment about why it is used here.

BFH-ktt1 replied:
The magic value has been replaced by ONLY_IO_MAP_NAME.
Done

MarcinOrlowski commented on src/main/java/com/cburch/logisim/fpga/data/MapComponent.java:
```diff
@@ -678,7 +680,7 @@ public String getDisplayString(int pin) {
   public void getMapElement(Element Map) throws DOMException {
     if (!hasMap()) return;
-    Map.setAttribute(MAP_KEY, getDisplayString(-1));
+    Map.setAttribute(MAP_KEY, getDisplayString(-2));
```
can we have magic value "-2" replaced with constant? If not or it makes no sense then please add the comment about why -2 is used here.

BFH-ktt1 replied:
The -2 has been replaced by ONLY_IO_MAP_NAME, done

BFH-ktt1 commented:
@MarcinOrlowski : Can you unblock or say what still needs to be done?

maehne reviewed (approved):
Looks good to me!

MarcinOrlowski commented on src/main/java/com/cburch/logisim/fpga/data/MapComponent.java:
```diff
@@ -678,7 +681,7 @@ public String getDisplayString(int pin) {
   public void getMapElement(Element Map) throws DOMException {
     if (!hasMap()) return;
-    Map.setAttribute(MAP_KEY, getDisplayString(-1));
+    Map.setAttribute(MAP_KEY, getDisplayString(ONLY_IO_MAP_NAME));
```
Formerly `getDisplayString()` was given `-1`. Not it is given `ONLY_IO_MAP_NAME` but that's `-2`. Is that correct?

BFH-ktt1 replied:
Yes, it should be like this, otherwise it get the display name with it like "-> Unconnected"

MarcinOrlowski commented:
@BFH-ktt1 I requested some changes that are marked as resolved but the change I see is not addressing my request at all. So either I miss something here, or GitHub shows it all in odd way or my comments were rejected. But as there's no comment there I am unable to figure out if the change made improves things or it's just what it was before, so unless I missed something I am not really willing to approve this PR because there's no counter argument nor explanation related to my change request.

BFH-ktt1 commented:
@MarcinOrlowski : I addressed your comments by replacing the -2 by ONLY_IO_MAP_NAME. So I do nit see how the change does not address your request ```can we have magic value "-2" replaced with constant``` could you elaborate?

MarcinOrlowski commented:
@BFH-ktt1  I see the use of constant and that's fine, thanks. But my I also noticed that  value of the constant differs from original "magic value" (`-1` vs `-2`). Is that correct?

BFH-ktt1 commented:
@MarcinOrlowski : Yes this is correct, with the -1 it will add for example "-> Unconnected" for display purposes, with -2 it will only show the "signal name" which is required for the map-purpose

MarcinOrlowski reviewed (approved):
OK, thanks for clarification.
